bobtimus/src/coin_selection.rs

```diff
@@ -0,0 +1,259 @@
+use anyhow::Result;
+use bdk::{
+    bitcoin::Amount,
+    database::{BatchOperations, Database},
+    wallet::coin_selection::CoinSelectionResult,
+    wallet::coin_selection::{BranchAndBoundCoinSelection, CoinSelectionAlgorithm},
+};
+use elements_fun::{ExplicitTxOut, OutPoint};
+
+#[derive(Clone, PartialEq)]
+pub struct Utxo {
+    outpoint: OutPoint,
+    txout: ExplicitTxOut,
+}
+
+impl From<Utxo> for bdk::UTXO {
+    fn from(utxo: Utxo) -> Self {
+        let value = utxo.txout.value.0;
+        let script_pubkey = utxo.txout.script_pubkey.into_bytes();
+        let script_pubkey = bdk::bitcoin::Script::from(script_pubkey);
+
+        Self {
+            outpoint: bdk::bitcoin::OutPoint {
+                txid: bdk::bitcoin::Txid::from_hash(utxo.outpoint.txid.as_hash()),
+                vout: utxo.outpoint.vout,
+            },
+            txout: bdk::bitcoin::TxOut {
+                value,
+                script_pubkey,
+            },
+            script_type: bdk::ScriptType::External,
+        }
+    }
+}
+
+pub fn coin_select(utxos: Vec<Utxo>, target: Amount) -> Result<Vec<Utxo>> {
```

Comment:
If we accept a `Utxo` here which has an `ExplicitTxOut`, we should probably assert that they are all the same asset. Alternatively, we could inline `value` and `script_pubkey` and document that it needs to be the same asset.